src/clean.c

```diff
@@ -73,8 +75,25 @@ char	*build_expanded_instruction(char *clean, char *instruction,
 	j = 0;
 	while (instruction[i])
 	{
-		if (instruction[i] == '$' && envir_must_be_expanded(instruction, i))
+		if (instruction[i] == '$' && (instruction[i + 1] == ' ' || instruction[i + 1] == '\0')) //ESTO DEBERIA SER CON  TABULADOR TAMBIEN???
+		{
```

Comment:
I think handling the edge case inside the function is not the proper way. We can have multiple edge cases that don't fit here. We must create a generic function that expands variables, like is now, and then another check that is run before the expansion to evaluate how to handle that case. As a first thought, we can evaluate using the token value and do normal expansion or edge case expansion